hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/placement/algorithms/SCMContainerPlacementRackAware.java

```diff
@@ -275,28 +277,32 @@ private Node chooseNode(List<Node> excludedNodes, Node affinityNode,
         throw new SCMException("No satisfied datanode to meet the" +
             " excludedNodes and affinityNode constrains.", null);
       }
-      if (super.hasEnoughSpace((DatanodeDetails)node, sizeRequired)) {
-        LOG.debug("Datanode {} is chosen. Required size is {}",
+
+      DatanodeInfo datanodeInfo = (DatanodeInfo) node;
+      NodeStatus nodeStatus = datanodeInfo.getNodeStatus();
+      if (nodeStatus.isNodeWritable() &&
+          (super.hasEnoughSpace(datanodeInfo, sizeRequired))) {
+        LOG.debug("Datanode {} is chosen. Required storage size is {} bytes",
             node.toString(), sizeRequired);
```

Comment:
`node.toString()` is going to always be executed even with INFO level logs. This should be changed to just `node` and let the logger call toString on it internally if debug is enabled.

Reply:
Good point.